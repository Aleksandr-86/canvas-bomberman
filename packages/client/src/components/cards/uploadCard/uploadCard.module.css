.inputFile {
  position: absolute;
  opacity: 0;
  visibility: hidden;
}

.buttonLink {
  display: block;
  padding: 10px 20px;
  border: none;
  background: none;
  color: blue;
  font-size: 25px;
  font-family: 'Rubik Bubbles', fantasy;
  text-transform: uppercase;
  cursor: pointer;
}
