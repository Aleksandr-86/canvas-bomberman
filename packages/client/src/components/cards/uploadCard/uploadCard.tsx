import React, { useState } from 'react'
import { Button } from '../../button/button'
import { Modal } from '../../modal/modal'

import cardStyles from '../card.module.css'
import inputStyles from '../../formField/formField.module.css'
import uploadStyles from './uploadCard.module.css'
import { useAppDispatch } from '../../../store/hooks'
import { uploadAvatar } from '../../../store/user/userActions'

interface Props {
  isActive?: boolean
  setActive?: any
}

export const UploadCard: React.FC<Props> = props => {
  const dispatch = useAppDispatch()
  const [label, setLabel] = useState('Файл не выбран')
  const [file, setFile] = useState({} as File | null)

  const { isActive, setActive } = props

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { files }: { files: FileList | null } =
      event.target as HTMLInputElement

    if (!files?.length) {
      return
    }

    const [_file] = files

    setFile(_file)
    setLabel(_file.name)
  }

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()

    if (!file || file === null) {
      return
    }

    const _file = new FormData()
    _file.append('avatar', file)

    dispatch(uploadAvatar(_file))

    setFile(null)
    setLabel('Файл не выбран')
  }

  return (
    <Modal isActive={isActive} setActive={setActive}>
      <form onSubmit={handleSubmit}>
        <div className={`${cardStyles.card} ${cardStyles.cardColumn}`}>
          <h3 className={cardStyles.cardHeader}>Загрузить файл</h3>

          <div className={cardStyles.cardContent}>
            <div className={inputStyles.label}>{label}</div>

            <input
              className={uploadStyles.inputFile}
              name="file"
              type="file"
              id="avatar"
              onChange={handleChange}
            />

            <label className={inputStyles.label} htmlFor="avatar">
              <span className={uploadStyles.buttonLink}>
                Выбрать с компьютера
              </span>
            </label>

            <p className={inputStyles.error}></p>
          </div>

          <div className={cardStyles.cardAction}>
            <Button type="submit">Загрузить</Button>
          </div>
        </div>
      </form>
    </Modal>
  )
}
