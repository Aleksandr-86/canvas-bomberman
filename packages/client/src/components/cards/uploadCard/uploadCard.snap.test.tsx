import * as renderer from 'react-test-renderer'
import { UploadCard } from './uploadCard'

describe(`<UploadCard/> snapshot test cases`, () => {
  test(`<UploadCard/> should render <Modal/> component with child <form> elements`, () => {
    const tree = renderer.create(<UploadCard />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
