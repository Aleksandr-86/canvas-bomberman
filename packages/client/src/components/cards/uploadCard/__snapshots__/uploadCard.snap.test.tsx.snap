// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadCard/> snapshot test cases <UploadCard/> should render <Modal/> component with child <form> elements 1`] = `
<div
  className="modal modalClose"
  onClick={[Function]}
>
  <div
    className="modalContent"
    onClick={[Function]}
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="card cardColumn"
      >
        <h3
          className="cardHeader"
        >
          Загрузить файл
        </h3>
        <div
          className="cardContent"
        >
          <div
            className="label"
          >
            Файл не выбран
          </div>
          <input
            className="inputFile"
            id="avatar"
            name="file"
            onChange={[Function]}
            type="file"
          />
          <label
            className="label"
            htmlFor="avatar"
          >
            <span
              className="buttonLink"
            >
              Выбрать с компьютера
            </span>
          </label>
          <p
            className="error"
          />
        </div>
        <div>
          <button
            className="button"
            type="submit"
          >
            Загрузить
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
